cpp/test/CMakeLists.txt

```diff
@@ -74,13 +86,34 @@ endif(BUILD_CUML_TESTS)
 # - build test_ml_mg executable ----------------------------------------------
 
 if(BUILD_CUML_MG_TESTS)
-  # (please keep the filenames in alphabetical order)
-  add_executable(ml_mg
-    mg/test_ml_mg_utils.cu)
+  find_package(MPI)
+
+  if(MPI_CXX_FOUND)
+    # (please keep the filenames in alphabetical order)
+    add_executable(ml_mg
+      mg/knn.cu
+      mg/main.cu
+      mg/pca.cu
+      mg/test_ml_mg_utils.cu)
+
+    add_dependencies(ml_mg cumlcommsmpi)
+
+    set(CUML_TEST_INCLUDE_DIRS
+        ${CUML_TEST_INCLUDE_DIRS}
+        ${MPI_CXX_INCLUDE_PATH}
+        ${CMAKE_CURRENT_SOURCE_DIR}/../comms/mpi/include)
+
+    target_include_directories(ml_mg PUBLIC ${CUML_TEST_INCLUDE_DIRS})
+
+    set(CUML_TEST_LINK_LIBRARIES ${CUML_TEST_LINK_LIBRARIES} ${MPI_CXX_LIBRARIES})
+
+    set(CUML_TEST_LINK_LIBRARIES ${CUML_TEST_LINK_LIBRARIES} cumlcommsmpi)
 
-  target_include_directories(ml_mg PRIVATE ${CUML_TEST_INCLUDE_DIRS})
+    target_link_libraries(ml_mg ${CUML_TEST_LINK_LIBRARIES})
 
-  target_link_libraries(ml_mg ${CUML_TEST_LINK_LIBRARIES})
+  else(MPI_CXX_FOUND)
+   message("OpenMPI not found. Skipping 'ml_mg'")
```

Comment:
Just occurred to me that users can build with MPI installed, but they are specifically going to need a CUDA-aware MPI (ideally w/ UCX). Can you create a Github issue to add that requirement to the docs? I recall there being a cuda-aware MPI conda package released, though I believe a special argument or flag needs to be passed in to enable it when installing the openmpi package.

Reply:
https://github.com/rapidsai/cuml/issues/2512